src/GovUk.Education.ExploreEducationStatistics.Notifier.Tests/Services/EmailServiceTests.cs

```diff
@@ -0,0 +1,227 @@
+        var values = new Dictionary<string, dynamic>
+        {
+            { "test-key", "test-value" },
+            { "test-key-2", 2 }
+        };
+
+        var notificationClient = new Mock<INotificationClient>(MockBehavior.Strict);
+
+        notificationClient
+            .Setup(s =>
+                s.SendEmail(
+                    email,
+                    templateId,
+                    values,
+                    null,
+                    null,
+                    null))
+            .Returns(new EmailNotificationResponse());
+
+        var service = BuildService(
+            notificationClient: notificationClient.Object);
+
+        service.SendEmail(
+            email: email,
+            templateId: templateId,
+            values: values);
+
+        notificationClient.Verify(
+            s => s.SendEmail(
+                email,
+                templateId,
+                values,
+                null,
+                null,
+                null)
+            , Times.Once);
+    }
+
+    [Fact]
+    public void SendEmail_GenericNotifyClientExceptionRaised_RethrowsException()
+    {
+        var email = "test@test.com";
+        var templateId = "test-template-id";
+        var values = new Dictionary<string, dynamic>
+        {
+            { "test-key", "test-value" },
+            { "test-key-2", 2 }
+        };
+
+        var notificationClient = new Mock<INotificationClient>(MockBehavior.Strict);
+
+        var originalException = new NotifyClientException(TestApiKeyErrorMessage);
+
+        notificationClient
+            .Setup(s =>
+                s.SendEmail(
+                    email,
+                    templateId,
+                    values,
+                    null,
+                    null,
+                    null))
+            .Throws(originalException);
+
+        var service = BuildService(
+            notificationClient: notificationClient.Object);
+
+        var exception = Assert.Throws<NotifyClientException>(() =>
+            service.SendEmail(
+                email: email,
+                templateId: templateId,
+                values: values));
+
+        notificationClient.Verify(
+            s => s.SendEmail(
+                email,
+                templateId,
+                values,
+                null,
+                null,
+                null)
+            , Times.Once);
+
+        Assert.Equal(originalException, exception);
+    }
+
+    [Fact]
+    public void SendEmail_TestApiNotifyClientExceptionRaised_SuppressionDisabled_RethrowsException()
+    {
+        var email = "test@test.com";
+        var templateId = "test-template-id";
+        var values = new Dictionary<string, dynamic>
+        {
+            { "test-key", "test-value" },
+            { "test-key-2", 2 }
+        };
+
+        var notificationClient = new Mock<INotificationClient>(MockBehavior.Strict);
+
+        var originalException = new NotifyClientException("Generic error");
```

Comment:
I think this should be `TestApiKeyErrorMessage` rather than "Generic error" as its in the test `TestApiNotifyClientExceptionRaised`.

Reply:
D'oh yeah well spotted!  Swapped thanks.